controllers/utils.go

```diff
@@ -202,3 +202,15 @@ func getCloudProviderFromInfra(c client.Client) (string, error) {
 
 	return strings.ToLower(string(infrastructure.Status.PlatformStatus.Type)), nil
 }
+
+// Method to get cluster id from ClusterVersion object
+func getClusterID(c client.Client) (string, error) {
+	clusterVersion := &configv1.ClusterVersion{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: "version"}, clusterVersion)
+	if err != nil {
+		return "", err
+	}
+
+	// Return first 8 characters of the cluster id
+	return string(clusterVersion.Spec.ClusterID[:8]), nil
+}
```

Comment:
Why not doing this directly in the scripts IIRC we do have kubectl/oc there available there

Reply:
You are right, this can be done in the scripts as well. The reason for doing it here is to feed the input config for gallery (azure-podvm-image-cm) via the operator. The azure-podvm-image-cm is input only config and I consciously avoided updating it in the script.